PowerEditor/src/WinControls/StaticDialog/StaticDialog.cpp

```diff
@@ -77,37 +77,111 @@ void StaticDialog::goToCenter()
 	::SetWindowPos(_hSelf, HWND_TOP, x, y, _rc.right - _rc.left, _rc.bottom - _rc.top, SWP_SHOWWINDOW);
 }
 
-void StaticDialog::display(bool toShow) const
+void StaticDialog::display(bool toShow, bool enhancedPositioningCheckWhenShowing) const
 {
 	if (toShow)
 	{
-		// If the user has switched from a dual monitor to a single monitor since we last
-		// displayed the dialog, then ensure that it's still visible on the single monitor.
-		RECT workAreaRect = {0};
-		RECT rc = {0};
-		::SystemParametersInfo(SPI_GETWORKAREA, 0, &workAreaRect, 0);
-		::GetWindowRect(_hSelf, &rc);
-		int newLeft = rc.left;
-		int newTop = rc.top;
-		int margin = ::GetSystemMetrics(SM_CYSMCAPTION);
-
-		if (newLeft > ::GetSystemMetrics(SM_CXVIRTUALSCREEN)-margin)
-			newLeft -= rc.right - workAreaRect.right;
-		if (newLeft + (rc.right - rc.left) < ::GetSystemMetrics(SM_XVIRTUALSCREEN)+margin)
-			newLeft = workAreaRect.left;
-		if (newTop > ::GetSystemMetrics(SM_CYVIRTUALSCREEN)-margin)
-			newTop -= rc.bottom - workAreaRect.bottom;
-		if (newTop + (rc.bottom - rc.top) < ::GetSystemMetrics(SM_YVIRTUALSCREEN)+margin)
-			newTop = workAreaRect.top;
-
-		if ((newLeft != rc.left) || (newTop != rc.top)) // then the virtual screen size has shrunk
-			// Remember that MoveWindow wants width/height.
-			::MoveWindow(_hSelf, newLeft, newTop, rc.right - rc.left, rc.bottom - rc.top, TRUE);
+		if (enhancedPositioningCheckWhenShowing)
+		{
+			RECT testPositionRc, candidateRc;
+
+			getWindowRect(testPositionRc);
+
+			candidateRc = getViewablePositionRect(testPositionRc);
+
+			if ((testPositionRc.left != candidateRc.left) || (testPositionRc.top != candidateRc.top))
+			{
+				::MoveWindow(_hSelf, candidateRc.left, candidateRc.top, 
+					candidateRc.right - candidateRc.left, candidateRc.bottom - candidateRc.top, TRUE);
+			}
+		}
+		else
+		{

```

Comment:
So this part could be replace by `getViewablePositionRect()` or I am missing somethings ?

Reply:
The `display` function is currently called for many of the different child windows.  The intent with this code change was to allow this enhanced positioning code to be introduced ONLY for the Find window (to stay in context of #905 and not have to validate ALL windows that use it -- too large of a change).  Other windows that can get "lost" by having positions outside viewable space would need to be handled one at a time (separate future "issues"), but in theory could use this new call option to display().  Once all child windows have been evaluated/changed to use the new code, the code I've moved to the "else" clause could be removed.

Reply:
Makes sens. Let's keep it in this PR.